CRM/Utils/System/Base.php

```diff
@@ -705,7 +705,7 @@ public function getTimeZoneOffset() {
 
       $timeZoneOffset = sprintf("%02d:%02d", $tz / 3600, abs(($tz / 60) % 60));
 
-      if ($timeZoneOffset > 0) {
+      if ($tz >= 0) { // $tz is a number, not a string, so it is better to test it.  00:00 needs a plus sign.
```

Comment:
But why hasn't it already returned at this 

   if (empty($tz)) {
        return FALSE;
      }